redev_variant_tools.h

```diff
@@ -0,0 +1,10 @@
+#ifndef REDEV__REDEV_VARIANT_TOOLS_H
+#define REDEV__REDEV_VARIANT_TOOLS_H
+namespace wdmcpl {
```

Comment:
I don't see this header being included in this repo's code.  If it is supposed to be used by applications and should be here as a helper, I'm a bit concerned about using the `wdmcpl` namespace in this repo.

IIUC, from our discussions, and scanning the code, this is needed to call `GetRank(...)` via a visit function that supports argument lists with different lengths and types (would it be correct to call it variadic?).  Is that correct?

Reply:
Yeah I noticed that I flubbed the namespace and I fixed it locally. Apparently I didn't push the change. This helper gives the convenience `overloaded` class which lets you do:

```c++
auto rank = std::visit(overloaded{ 
[](RCBPartition& p){p.GetRank(...);},
[](ClassPartition& p){p.GetRank(...);},
},partition);
```
instead of 
```c++
struct GetRankFunctor
{
operator()(RCBPartition& p) {p.GetRank(...);}
operator()(ClassPartition& p) {p.GetRank(...);}
}
std::visit(GetRankFunctor{}, partition);
```